File: shoppingm8_fe/lib/auth/dto/authenticatedUserDto.dart
class AuthenticatedUserDto {
  final num id;
  final String name;
  final String email;
  final String profilePicture;

  AuthenticatedUserDto({this.id, this.name, this.email, this.profilePicture});
  
  factory AuthenticatedUserDto.fromJson(Map<String, dynamic> json) {
    return AuthenticatedUserDto(
        id: json['id'],
        name: json['name'],
        email: json['email'],
        profilePicture: json['profilePicture']
    );
  }
}

File: ShoppingM8-be/src/main/java/edu/pw/shoppingm8/authentication/api/dto/AuthenticatedUserDto.java
package edu.pw.shoppingm8.authentication.api.dto;

import edu.pw.shoppingm8.user.User;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class AuthenticatedUserDto {
    private static final String USER_PICTURE_URL_TEMPLATE = "/user/{id}/picture";

    Long id;
    String name;
    String email;
    String profilePicture;

    public static AuthenticatedUserDto of(User user) {
        return AuthenticatedUserDto.builder()
                .id(user.getId())
                .name(user.getName())
                .email(user.getEmail())
                .profilePicture(buildProfilePictureUrl(user.getId()))
                .build();
    }

    private static String buildProfilePictureUrl(Long id) {
        return String.format(USER_PICTURE_URL_TEMPLATE, id);
    }
}
